File: SDKs/amazonaws.com/datapipeline/2012-10-29/java/lib/src/main/java/openapisdk/models/shared/ListPipelinesOutput.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ListPipelinesOutput
 * Contains the output of ListPipelines.
**/
public class ListPipelinesOutput {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("hasMoreResults")
    public Boolean hasMoreResults;
    public ListPipelinesOutput withHasMoreResults(Boolean hasMoreResults) {
        this.hasMoreResults = hasMoreResults;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("marker")
    public String marker;
    public ListPipelinesOutput withMarker(String marker) {
        this.marker = marker;
        return this;
    }
    @JsonProperty("pipelineIdList")
    public PipelineIdName[] pipelineIdList;
    public ListPipelinesOutput withPipelineIdList(PipelineIdName[] pipelineIdList) {
        this.pipelineIdList = pipelineIdList;
        return this;
    }
}

File: SDKs/amazonaws.com/datapipeline/2012-10-29/python/src/sdk/models/shared/listpipelinesoutput.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import pipelineidname as shared_pipelineidname


@dataclass_json
@dataclasses.dataclass
class ListPipelinesOutput:
    r"""ListPipelinesOutput
    Contains the output of ListPipelines.
    """
    
    pipeline_id_list: list[shared_pipelineidname.PipelineIDName] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('pipelineIdList') }})
    has_more_results: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('hasMoreResults') }})
    marker: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('marker') }})
    
